### ISSUE TITLE
[ISSUE] Changing permission results in permission removals

### ISSUE THREAD

**Issue by Gnarik** [databrickslabs/terraform-provider-databricks#309]
Hi there,

### Terraform Version
Terraform v0.12.29

### Affected Resource(s)
Please list the resources as a list, for example:
- databricks_permissions

### Terraform Configuration Files
```hcl
provider "databricks" {
  host = "ENDPOINT"
  token = "TOKEN"
}

resource "databricks_cluster" "test_cluster" {
  cluster_name = "test-cluster-permission"
  spark_version = "7.1.x-scala2.12"
  node_type_id = "i3.xlarge"
  autotermination_minutes = 30

  autoscale {
    min_workers = 1
    max_workers = 2
  }

  aws_attributes {
    zone_id = "us-east-1a"
    availability = "SPOT_WITH_FALLBACK"
    spot_bid_price_percent = 100
    instance_profile_arn = "arn:aws:iam::xxxxxxxxxxx:instance-profile/rolename"
    first_on_demand = "1"
  }

}

resource "databricks_permissions" "test-permission" {
  cluster_id = databricks_cluster.test_cluster.id
  access_control {
    user_name = "runtime-user"
    permission_level = "CAN_ATTACH_TO"
  }
}
```
At this point, `run-time` user has `CAN_ATTACH_TO` permission on the cluster. Then I run:
```hcl
provider "databricks" {
  host = "ENDPOINT"
  token = "TOKEN"
}

resource "databricks_cluster" "test_cluster" {
  cluster_name = "test-cluster-permission"
  spark_version = "7.1.x-scala2.12"
  node_type_id = "i3.xlarge"
  autotermination_minutes = 30

  autoscale {
    min_workers = 1
    max_workers = 2
  }

  aws_attributes {
    zone_id = "us-east-1a"
    availability = "SPOT_WITH_FALLBACK"
    spot_bid_price_percent = 100
    instance_profile_arn = "arn:aws:iam::xxxxxxxxxxx:instance-profile/rolename"
    first_on_demand = "1"
  }

}

resource "databricks_permissions" "test-permission" {
  cluster_id = databricks_cluster.test_cluster.id
  access_control {
    user_name = "runtime-user"
    permission_level = "CAN_MANAGE"
  }
}
```

### Debug Output
https://gist.github.com/Gnarik/49fcfcc5d24c376c2d3b12b294c1c379

### Expected Behavior
I expect my runtime-user to get new level of permission: from `CAN_ATTACH_TO` to `CAN_MANAGE`

### Actual Behavior
All users which have any permissions on the cluster got removed from the cluster permission list.

### Steps to Reproduce
Please list the steps required to reproduce the issue, for example:
1. `terraform apply`

**Comment by nfx** [comment]
@Gnarik please update to `0.2.4`, as you're using old version of provider. Please confirm if issue still persists.

According to standard Terraform behaviour, if state is set outside of terraform, it would be overridden. E.g. if you want to manage permissions for users/groups - it should be done exclusively from Terraform .

this might be related to #303

**Comment by Gnarik** [comment]
Hi,

Thank you for your answer.

I got the same exact error with 0.2.4 and terraform 0.13 : https://gist.github.com/Gnarik/c544b2f56d40d625683119047240fd70

The only difference in my main.tf is 
```hcl
terraform {
  required_providers {
    databricks = {
      source = "databrickslabs/databricks"
      version = "0.2.4"
    }
  }
}
```

**Comment by nfx** [comment]
@Gnarik thanks for confirmation, then it's a duplicate of #303 , closing this one and let's track the progress in earlier issue.

**Comment by nfx** [comment]
@Gnarik released in 0.2.7

**Comment by romaric-sticky** [comment]
Hi, thank you for letting me know.
